keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,727 @@
+    //
+    // +required
+    CIDR string `json:"cidr,omitempty" protobuf:"bytes,1,rep,name=cidr"`
+}
+```
+
+### User Stories (Optional)
+
+#### Story 1 - Network Device Status for Network Services
+
+As a Cloud Native Network Function (CNF) vendor, my network services must be
+integrated with the network devices configured in Pods. The configuration
+properties of these network devices are therefore essential to configure the
+network services. For example, the network services must be able to route
+traffic to pods over networks attached via the network devices, the IP
+addresses of the network device(s) must then be reflected in the
+`ResourceClaim.Status` allowing the network service controller(s) to access
+them.
+
+#### Story 2 - Network Device Status for Troubleshooting
+
+As a Network Administrator, troubleshooting networking issues can be complex
+and time consuming especially when the device characteristics and operational
+status are not readily accessible. The `Devices` field in the
+`ResourceClaim.Status` provides access to comprehensive details regarding
+network interfaces helping to quickly and efficiently identify the issues such
+as error messages on failed network interface configuration, incorrect IP
+assignments or misconfigured network interfaces.
+
+### Notes/Constraints/Caveats (Optional)
+
+The content of `Data` is driver specific and not standardized as part of
+DRA, the interpretation of this field may then vary between controllers and
+users reading it. 
+
+The accuracy of the information depends on the implementation of the DRA
+Drivers, the reported status of the device may not always reflect the real time
+changes of the state of the device.
+
+### Risks and Mitigations
+
+As stated, 3rd party DRA drivers will set and update the `Devices` for
+the device they manage. An access control must be set in place to restrict the
+write access to the appropriate driver (A device status can only be updated by
+the entities that have a direct control over the device(s) being reported).
+
+Adding `Data` as an arbitrary data slice may introduce extra processing
+and storage overhead which might impact performance in a cluster with many
+devices and frequent status updates. In large-scale clusters where many devices
+are allocated, this impact must be considered.
+
+## Design Details
+
+### API
+
+The `ResourceClaimStatus` struct in `pkg/apis/resource/types.go` will be
+extended to include the slice of `Devices`.
+
+`ResourceClaim` validation of the status in
+`pkg/apis/resource/validation/validation.go` will be covered to allow a device
+to be reported only once in the slice, a device is being identified  by
+`<driver name>/<pool name>/<device name>`.
+
+### Write Permission
+
+To prevent unauthorized or accidental modifications by entities that do not 
+have access to a particular resource, a `ValidatingAdmissionPolicy` will be 
+created to validate the entities attempting to update the devices in the 
```

Comment:
just curios, is this created at cluster startup? using the `plugin/pkg/auth/authorizer`?

Reply:
No, this is something that a cluster admin has to add once the cluster is up and running.

I have a topic pending for SIG Arch about automatically managing VAPs that are part of core Kubernetes. But as for CRDs, there's probably not going to be a good solution.